linode/manager
# chore: [m3-6511] Changelog Automation

## Description 📝
Here we go! This PR introduces an automated process to:
- generate changesets
- generate a changelog entry based on existing changeset files

changesets get created (and optionally committed automatically) via the command line, along with a type ('Added', 'Fixed', 'Changed', 'Removed', 'Tech Stories') and a reference to the PR (automatically via the GitHub CLI). You can write, add and commit changesets manually as well if you so wish.

changelog entry also get created through the command line, prompting for a date and a semver variant.

once the script has ran successfully, changesets associated with the release are deleted. Both changes in `CHANGELOG.md` and deleted associated changeset must be committed manually, mostly due to our current release process (we can always automate more in the future!).

In its current state **this PR modifies nothing to our setup**. This is intentional as we aim to do a try run. a new `CHANGELOG.md` file (with the current content) has been added at the root of @package/manager (where it should only leave eventually). Eventually we will want to:
- delete the changelog file at the monorepo's root
- delete the `generate_changelog.py`, remove its call and get rid of all py deps ✌️ 
- update release instruction 
- re-think our PRs and commits naming conventions? (do we still need to parse those)

## Preview 📷
https://github.com/linode/manager/assets/130582365/9399a111-5d85-439f-819c-466c73d54667

## How to test 🧪
1. Pull code locally and run `yarn`
2. If you don't have it, install GH CLI: https://cli.github.com/manual/installation (it's an amazing tool, not just for this!)
3. Once installed, run `gh repo set-default` and pick `linode/manager` (important step since we are using forks, but the branch we'll be looking for points to `linode/manager`)
4. Create and open a new test branch (we'll always assume that you have a branch open when adding a changeset) - It will yell at you if you don't anyway
5. run `yarn changeset` and do your thing, as many times as you want
6. run `yarn generate-changelog` and check the `CHANGELOG.md` at @package/manager (not the root one)
7. Hopefully that worked out!
